src/test/java/cloud/fogbow/ras/core/plugins/interoperability/opennebula/volume/v5_4/OpenNebulaVolumePluginTest.java

```diff
@@ -1,343 +1,393 @@
 	}
-	
-	// test case: When calling the requestInstance method, with a valid client and a
-	// request with a volume name, a template must be generated with the associated
-	// data, to allocate a volume, returning its instance ID.
+
+	// test case: When calling the doRequestInstance method, with valid create volume request and client,
+	// the plugin should call getDataStoreId to discover where to store the volume and then allocate
+	// the new volume with the request template.
 	@Test
-	public void testRequestInstanceSuccessfullyWithoutDefaultVolumeName() throws FogbowException {
+	public void testDoRequestInstance() throws UnexpectedException, InvalidParameterException, NoAvailableResourcesException {
 		// set up
-		Client client = Mockito.mock(Client.class);
-		PowerMockito.mockStatic(OpenNebulaClientUtil.class);
-		BDDMockito.given(OpenNebulaClientUtil.createClient(Mockito.anyString(), Mockito.anyString()))
-				.willReturn(client);
+		int datastoreId = Integer.parseInt(STRING_VALUE_ONE);
+		CreateVolumeRequest createVolumeRequest = Mockito.spy(this.createVolumeRequest(this.volumeOrder));
+		Mockito.doReturn(datastoreId).when(this.plugin).getDataStoreId(Mockito.any(Client.class), Mockito.anyLong());
 
-		String template = generateImageTemplate(1);
-		String instanceId = STRING_VALUE_ONE;
+		// exercise
+        this.plugin.doRequestInstance(createVolumeRequest, this.client);
 
-		BDDMockito.given(OpenNebulaClientUtil.allocateImage(Mockito.eq(client), Mockito.eq(template), Mockito.anyInt()))
-				.willReturn(instanceId);
+		// verify
+		PowerMockito.verifyStatic(OpenNebulaClientUtil.class, Mockito.times(TestUtils.RUN_ONCE));
+		OpenNebulaClientUtil.allocateImage(Mockito.any(Client.class), Mockito.eq(this.template), Mockito.eq(datastoreId));
 
-		OneResponse response = Mockito.mock(OneResponse.class);
-		PowerMockito.mockStatic(Image.class);
-		BDDMockito.given(Image.allocate(Mockito.eq(client), Mockito.eq(template), Mockito.anyInt()))
-				.willReturn(response);
-		Mockito.when(response.isError()).thenReturn(false);
-		Mockito.when(response.getMessage()).thenReturn(instanceId);
+		Mockito.verify(createVolumeRequest, Mockito.times(TestUtils.RUN_ONCE)).getVolumeImage();
+		Mockito.verify(this.plugin, Mockito.times(TestUtils.RUN_ONCE)).getDataStoreId(
+				Mockito.any(Client.class), Mockito.eq(Long.parseLong(DISK_VALUE_30GB)));
+	}
+
+	// test case: when calling doRequestInstance and receiving a null value (meaning no datastore with enough available
+	// free storage was found) from the getDataStoreId call, then a NoAvailableResources exception should be thrown
+	@Test
+	public void testDoRequestInstanceFail() throws UnexpectedException, InvalidParameterException {
+		// set up
+		CreateVolumeRequest createVolumeRequest = Mockito.spy(this.createVolumeRequest(this.volumeOrder));
+		Mockito.doReturn(null).when(this.plugin).getDataStoreId(Mockito.any(Client.class), Mockito.anyLong());
+
+		// exercise
+        try {
```

Comment:
the indentation of this test seems weird

Reply:
i attempted to fix it, let's see if it works. ;-)